src/main/scala/org/deepdive/ddlog/DeepDiveLogSemanticChecker.scala

```diff
@@ -142,6 +147,43 @@ object DeepDiveLogSemanticChecker extends DeepDiveLogHandler {
     }
   }
 
+  // check if head variables have bindings in the body
+  def checkHeadVariables(stmt: Statement) {
```

Comment:
It's great to have this kind of extra checks.  Ideally, we should check in every rule whether all variables used in expressions are bound by at least one pattern.  Checking the head only covers part of that general requirement.  This should be much clearer once we separate the `Patt` type from `Expr`s.